### PR TITLE
github: update linux runners

### PR DESCRIPTION
**Pull request by tamiko**
- bump all runners to Ubuntu 22.04. It's time.
 - unify output between linux runners, enable quick_tests and example tests

### CODE REVIEW

**Review by drwells — APPROVED** [review]

**Comment by tamiko** [comment]
@tjhei *ping*

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -16,44 +17,35 @@ permissions:
...
         g++ -v
         cmake --version
-    - name: modules
-      run: |
-        sudo apt-get install g++-10
-    - name: Build Boost
-      id: boost
-      uses: egor-tensin/build-boost@v1
-      with:
-        version: 1.74.0
```
Comment (tjhei) [inline .github/workflows/linux.yml:44]: Is there a particular reason to use this boost version? What is in the image?

**Review by tjhei — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 1 above)
Comment (marcfehling) [reply .github/workflows/linux.yml:44]: There were C++20 issues with out bundled boost version, see https://github.com/dealii/dealii/pull/13535#discussion_r829365480. It seems like they no longer manifest in `22.04`.

**Comment by tamiko** [comment]
Alright. This is a bit obscure.

**Comment by bangerth** [comment]
Well, the Python bindings fail:
```
Start 22: python-bindings/triangulation_wrapper
21/22 Test #22: python-bindings/triangulation_wrapper ...***Failed    0.05 sec
Traceback (most recent call last):
  File "/__w/dealii/dealii/contrib/python-bindings/tests/triangulation_wrapper.py", line 17, in <module>
    from PyDealII.Debug import *
ModuleNotFoundError: No module named 'PyDealII.Debug'

22/22 Test #16: quick_tests/umfpack.release .............   Passed   10.33 sec
Errors while running CTest

73% tests passed, 6 tests failed out of 22

Total Test time (real) =  63.[83](https://github.com/dealii/dealii/actions/runs/5460000825/jobs/9936541714?pr=15652#step:8:84) sec

The following tests FAILED:
	 17 - python-bindings/cell_accessor_wrapper (Failed)
	 18 - python-bindings/manifold_wrapper (Failed)
	 19 - python-bindings/mapping_wrapper (Failed)
	 20 - python-bindings/point_wrapper (Failed)
	 21 - python-bindings/quadrature_wrapper (Failed)
	 22 - python-bindings/triangulation_wrapper (Failed)
```

**Comment by tamiko** [comment]
@bangerth I was staring at this for an hour straight. The key is to realize that the tests try to import `Debug`... we only build release...

**Comment by bangerth** [comment]
Well, I guess I would have missed that even after an hour :-)

**Comment by tamiko** [comment]
@drwells @tjhei @bangerth @marcfehling This needs reviews now :-)

**Review by tjhei — APPROVED** [review]

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -93,16 +95,16 @@ jobs:
         mkdir build
         cd build
         cmake -D CMAKE_BUILD_TYPE=Debug \
-              -D DEAL_II_CXX_FLAGS='-Werror -std=c++17' \
+              -D DEAL_II_CXX_FLAGS='-Werror -std=c++20' \
               -D DEAL_II_EARLY_DEPRECATIONS=ON \
-              -D DEAL_II_WITH_CGAL="ON" \
+              -D DEAL_II_COMPONENT_EXAMPLES="OFF" \
```
Comment (masterleinad) [inline .github/workflows/linux.yml:100]: Do you know why we are turning examples off?

**Review by masterleinad — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 2 above)
Comment (tamiko) [reply .github/workflows/linux.yml:100]: Beats me. @tjhei's plan is to convert all example steps into full fledged tests under `tests/examples` anyway over the next weeks so I simply left it disabled.

**Pull request merged**